src/include/metrics/query_trace_metric.h

```diff
@@ -111,10 +125,24 @@ class QueryTraceMetric : public AbstractMetric<QueryTraceMetricRawData> {
   friend class MetricsStore;
 
   void RecordQueryText(const execution::query_id_t query_id, const std::string &query_text, const uint64_t timestamp) {
-    GetRawData()->RecordQueryText(query_id, query_text, timestamp);
+    GetRawData()->RecordQueryText(query_id, "\"" + query_text + "\"", timestamp);
   }
-  void RecordQueryTrace(const execution::query_id_t query_id, const uint64_t timestamp) {
-    GetRawData()->RecordQueryTrace(query_id, timestamp);
+  void RecordQueryTrace(const execution::query_id_t query_id, const uint64_t timestamp,
+                        common::ManagedPointer<const std::vector<parser::ConstantValueExpression>> param,
+                        catalog::db_oid_t oid) {
+    std::ostringstream param_stream;
+    std::ostringstream type_stream;
+    for (const auto &val : (*param)) {
+      if (val.IsNull()) {
+        return;
```

Comment:
We just bail if a parameter is NULL?

Reply:
The null valued parameters are observed to only come from the database loading phase. For the pilot's purpose, we only need the queries from the execution phase. It would be better if we can directly detect if the query is from the loading phase and ignore it. But I haven't found a way to do so.

Reply:
That sounds like a design decision based on oltpbench. We want to solve the general case. Couldn't query trace metrics be turned on for any arbitrary workload?

Reply:
Maybe we can just record `NULL` here, since that's how postgres specifies NULL value in a SQL query.

Reply:
Fixed in the latest push.